### ISSUE TITLE
GLU bottleneck 

### ISSUE THREAD

**Issue by YimengZhu** [hirofumi0810/neural_sp#8]
Hi, I have a question regarding your GLU implementation [here](https://github.com/hirofumi0810/neural_sp/blob/master/neural_sp/models/modules/glu.py). 

As far as I understood the GCNN paper, the gate mechanism should be taken place in bottleneck layer if it is issued. In your implementation, however, it is happened after resample the bottleneck to output dimension.
`            layers['conv_out'] = nn.utils.weight_norm(
                nn.Conv2d(in_channels=bottlececk_dim,
                          out_channels=out_ch * 2,
                          kernel_size=(1, 1)), name='weight', dim=0)`

I'm not sure if I really understood the paper correctly, I would have implemented it like this:
` 
 def __init__(...):
        .....
        elif bottlececk_dim > 0:
            layers['conv_in'] = nn.utils.weight_norm(
                nn.Conv2d(in_channels=in_ch,
                          out_channels=bottlececk_dim,
                          kernel_size=(1, 1)), name='weight', dim=0)
            layers['dropout_in'] = nn.Dropout(p=dropout)
            layers['layers'] = nn.utils.weight_norm(
                nn.Conv2d(in_channels=bottlececk_dim,
                          out_channels=bottlececk_dim * 2,
                          kernel_size=(kernel_size, 1)), name='weight', dim=0)
            layers['dropout'] = nn.Dropout(p=dropout)

        self._layers1 = nn.Sequential(layers)
        self._layers2 = nn.Sequential(
               nn.utils.weight_norm(
               nn.Conv2d(in_channels=bottlececk_dim,
                          out_channels=out_ch,
                          kernel_size=(1, 1)), name='weight', dim=0)
            layers['dropout_out'] = nn.Dropout(p=dropout)
        )

def forward(...):
       ......#padding, residual etc.
       x = self._layers1(x)
       x = F.glu(x)
       x = self._layers2(x)
       .......
`
